### PR TITLE
ci(jenkins): disable stages for only docs

### PR DESCRIPTION
**Pull request by v1v**
### What

- Any builds for any branch or PR that only contains changes related to the asciidoc then it will skip all the stages but the `checkout`

### Why

Docs validation happens with the `elasticsearch-ci/docs` therefore there is no need to validate anything in our end.

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -53,6 +53,9 @@ pipeline {
               "^test/instrumentation/modules/"
             ]
             env.TAV_UPDATED = isGitRegionMatch(patterns: regexps)
+
+            // Skip all the stages except docs for PR's with asciidoc changes only
+            env.ONLY_DOCS = isGitRegionMatch(patterns: [ '.*\\.asciidoc' ], comparator: 'regexp', shouldMatchAll: true)
```
Comment (watson) [inline .ci/Jenkinsfile]: We should include `.md` as well

Comment (v1v) [reply .ci/Jenkinsfile]: are those files hosted in the same folder?

Comment (watson) [reply .ci/Jenkinsfile]: No, all the markdown files are currently in the root of the repo (where there's also one asciidoc file). The rest of the asciidoc files are all in the `docs` directory). I thought this pattern check was for all files no matter where in the repo they were located?

Comment (watson) [reply .ci/Jenkinsfile]: Come to think of it, for markdown changes, there's no reason to run any CI checks at all (besides to check if the committer have signed the CLA). For asciidoc changes, we should still run the `elasticsearch/docs` step

Comment (v1v) [reply .ci/Jenkinsfile]: got it, I'll then use a pattern like `'.*\\.(asciidoc|md)'` then all the changes in the PR should be markdown or asciidoc files.

**Pull request merged**